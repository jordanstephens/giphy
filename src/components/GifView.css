.gif {
  display: block;
  height: auto;
  width: 100%;
}

.gif-info {
  display: grid;
  grid-template-columns: 180px 1fr;
  font-size: 14px;
  line-height: 1.6;
  background-color: rgba(255, 255, 255, 0.25);
  padding: 20px;
}

.gif-info dt { font-weight: bold; }
.gif-info dd { margin-left: 10px; }

.gif-info a { color: #fff; }
