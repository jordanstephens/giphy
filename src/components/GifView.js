import React from 'react';
import PropTypes from 'prop-types';
import filesize from 'filesize';
import configuration from '../configuration';

import './GifView.css';

const GifView = ({ gif }) => {
  if (!gif) return null;

  const image = gif.images.original;
  const colors = configuration.colors;
  const backgroundColor = colors[gif.id.charCodeAt('0') % colors.length];

  return (
    <section>
      <div>
        <header>
          <h2>{gif.title}</h2>
        </header>
        <main>
          <picture>
            <source srcSet={image.webp} type="image/webp" />
            <img
              src={image.url}
              alt={gif.title}
              height={image.height}
              width={image.width}
              style={{ backgroundColor }}
              className='gif'
            />
          </picture>
        </main>
      </div>
      <footer>
        <dl className='gif-info'>
          <dt>Source</dt>
          <dd>
            <a
              href={gif.source}
              target='_blank'
              rel='noopener noreferrer'
            >
              {gif.source_tld}
            </a>
          </dd>
          <dt>Rating</dt>
          <dd>{gif.rating.toUpperCase()}</dd>
          <dt>Uploaded</dt>
          <dd>{new Date(gif.import_datetime).toLocaleDateString()}</dd>
          <dt>Dimensions</dt>
          <dd>{image.width} x {image.height} px</dd>
          <dt>Size</dt>
          <dd>{filesize(image.size)}</dd>
          <dt>Frames</dt>
          <dd>{image.frames}</dd>
        </dl>
      </footer>
    </section>
  );
};

GifView.propTypes = {
  gif: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    rating: PropTypes.string.isRequired,
    source: PropTypes.string.isRequired,
    source_tld: PropTypes.string.isRequired,
    import_datetime: PropTypes.string.isRequired,
    images: PropTypes.shape({
      original: PropTypes.shape({
        webp: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired,
        width: PropTypes.number.isRequired,
        height: PropTypes.number.isRequired,
        size: PropTypes.string.isRequired,
        frames: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  }),
};

export default GifView;
