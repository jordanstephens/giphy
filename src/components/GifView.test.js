/*global expect*/
import * as fs from 'fs';
import React from 'react';
import GifView from './GifView';
import renderer from 'react-test-renderer';

const fixture = JSON.parse(fs.readFileSync('src/services/fixtures/gif.json'));

function render(props = {}) {
  return renderer.create(
    <GifView {...props} />
  ).toJSON();
}

it('renders without a gif', () => {
  const tree = render({});
  expect(tree).toMatchSnapshot();
});

it('renders with a gif', () => {
  const tree = render({ gif: fixture.data });
  expect(tree).toMatchSnapshot();
});
