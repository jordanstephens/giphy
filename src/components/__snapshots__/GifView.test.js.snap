// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a gif 1`] = `
<section>
  <div>
    <header>
      <h2>
        
      </h2>
    </header>
    <main>
      <picture>
        <source
          srcSet="https://media2.giphy.com/media/H3qWY13eGZUomKxOs6/giphy.webp?cid=b430c6c65d1fdd9a376a726d49a8db49&rid=giphy.webp"
          type="image/webp"
        />
        <img
          alt=""
          className="gif"
          height="267"
          src="https://media2.giphy.com/media/H3qWY13eGZUomKxOs6/giphy.gif?cid=b430c6c65d1fdd9a376a726d49a8db49&rid=giphy.gif"
          style={
            Object {
              "backgroundColor": "rgb(153, 251, 255)",
            }
          }
          width="480"
        />
      </picture>
    </main>
  </div>
  <footer>
    <dl
      className="gif-info"
    >
      <dt>
        Source
      </dt>
      <dd>
        <a
          href="https://youtu.be/QJ0rDnoi8A4"
          rel="noopener noreferrer"
          target="_blank"
        >
          youtu.be
        </a>
      </dd>
      <dt>
        Rating
      </dt>
      <dd>
        G
      </dd>
      <dt>
        Uploaded
      </dt>
      <dd>
        2019-7-5
      </dd>
      <dt>
        Dimensions
      </dt>
      <dd>
        480
         x 
        267
         px
      </dd>
      <dt>
        Size
      </dt>
      <dd>
        1.27 MB
      </dd>
      <dt>
        Frames
      </dt>
      <dd>
        35
      </dd>
    </dl>
  </footer>
</section>
`;

exports[`renders without a gif 1`] = `null`;
